src/vendor/errors.jsx

```diff
@@ -1,154 +1,115 @@
-      <span>
-        There has been a critical error. We have reported it. If the issue is
-        not fixed within few hours please file an issue:
-        <br />
-        <a href={newIssue} target="_blank" rel="noopener noreferrer">
-          {newIssue}
-        </a>
-      </span>
-    </p>
-  );
+import SETTINGS from './settings';
+
+const RED = SETTINGS.colors.error;
+const styles = {
+  container: {
+    position: 'relative',
+    height: '100%',
+    width: '100%',
+  },
+  frame: {
+    boxSizing: 'border-box',
+    position: 'absolute',
+    top: 0,
+    bottom: 0,
+    left: 0,
+    right: 0,
+    borderStyle: 'solid',
+    borderWidth: '0.2rem',
+    borderColor: RED,
+  },
+  message: {
+    boxSizing: 'border-box',
+    backgroundColor: RED,
+    color: 'white',
+    width: '100%',
+    position: 'absolute',
+    margin: 0,
+    padding: 0,
+    bottom: 0,
+    height: '1.0rem',
+    textAlign: 'center',
+  },
 };
 
-class Exception extends Error {
-  constructor(template, params, cause) {
-    super();
-    let t = null;
-    let c = null;
-    let p = null;
-
-    if (template instanceof Exception || template instanceof Error) {
-      c = template;
-    } else if (
-      missing(cause) &&
-      (params instanceof Exception || params instanceof Error)
-    ) {
-      t = template;
-      c = params;
-    } else {
-      t = template;
-      p = params;
-      c = cause ? cause.toString() : '';
-    }
-
-    this.template = t; // string descripbing the problem
-    this.props = p; // object with properties used by template
-    this.cause = c; // chained reason
-  }
-
-  toString() {
-    if (missing(this.template)) {
-      return this.cause.toString();
-    }
-
-    let output = expand(this.template, this.props);
-
-    if (this.cause) {
-      output = `${output}\ncaused by\n${this.cause.toString()}`;
-    }
-
-    return output;
+class RawErrorMessage extends React.Component {
+  render() {
+    const {
+      error,
+      classes: { container, frame, message },
+      children,
+    } = this.props;
+
+    return (
+      <div className={container}>
+        {children}
+        <div className={frame} />
+        <div className={message}>{error.message}</div>
+      </div>
+    );
   }
 }
 
-Exception.wrap = err => {
-  if (err instanceof Exception) {
-    return err;
-  }
-
-  return new Exception(err);
-};
-
-class Log {}
-
-Log.error = (template, params, cause) => {
-  throw new Exception(template, params, cause);
-};
-
-Log.warning = (template, params, cause) => {
-  const e = new Exception(template, params, cause);
-
-  // eslint-disable-next-line no-console
-  console.log(e.toString());
-};
-
-class ErrorMessage extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = {};
+const ErrorMessage = withStyles(styles)(RawErrorMessage);
+const withErrorBoundary = WrappedComponent => {
+  if (
+    WrappedComponent.displayName &&
+    WrappedComponent.displayName.startsWith('WithStyles')
+  ) {
```

Comment:
`componentDidMount()` is a function that returns `undefined`.  `withStyles` returns a component that does just that. 

This is problem because we use `componentDidMount()` to launch any number of promise chains that may fail sometime.  Our components use `async componentDidMount()` which is just a function the returns a promise; React ignores the returned promise, but that does not stop the promise chain from running.

The `withErrorBoundary` is creating a class that uses the promise we are returning from `async componentDidMount()` and ensuring raised errors are properly caught and recorded.

Therefore, 

    export default withErrorBoundary(withStyles(styles)(PerfherderGraphContainer));

is not allowed